client/src/Pages/Dallas.js

```diff
@@ -0,0 +1,226 @@
+import { useState, useEffect } from "react";
+
+const Dallas = (props) => {
+  // Initial post in case that you want to update a new post
+  const {
+    initialPost = {
+      id: null,
+      username: "",
+      imageurl: "",
+      alt: "",
+      dish: "",
+      restaurant: "",
+      content: "",
+      city: "",
+      date: "",
+    },
+  } = props;
+
+  // We're using that initial student as our initial state
+  const [post, setPost] = useState(initialPost);
+
+  const [restaurants, setRestaurants] = useState({alias: "", name: "", address: ""});
+
+  //create functions that handle the event of the user typing into the form
+  const handleUsernameChange = (event) => {
+    const username = event.target.value;
+    setPost((post) => ({ ...post, username }));
+  };
+
+  const handleImageURLChange = (event) => {
+    const imageurl = event.target.value;
+    setPost((post) => ({ ...post, imageurl }));
+  };
+
+  const handleAltChange = (event) => {
+    const alt = event.target.value;
+    setPost((post) => ({ ...post, alt }));
+  };
+
+  const handleDishChange = (event) => {
+    const dish = event.target.value;
+    setPost((post) => ({ ...post, dish }));
+  };
+
+  const handleRestaurantChange = (event) => {
+    const restaurant = event.target.value;
+    setPost((post) => ({ ...post, restaurant }));
+  };
+
+  const handleContentChange = (event) => {
+    const content = event.target.value;
+    setPost((post) => ({ ...post, content }));
+  };
+
+  const handleCityChange = (event) => {
+    const city = event.target.value;
+    setPost((post) => ({ ...post, city }));
+  };
+
+  const handleDateChange = (event) => {
+    const date = event.target.value;
+    setPost((post) => ({ ...post, date }));
+  };
+
+  function getRestaurants(e) {
+    e.preventDefault();
+    fetch("/api/location-search", {
+      method: "GET",
+      headers: {"Content-Type": "application/json",},
+    }) .then((response) => response.json())
+    .then((data) => {
+      console.log("data check line 13 ", data.businesses[0]);
+      let fetchRestaurant = {
+        alias: data.businesses[0].alias,
+        name: data.businesses[0].name,
+        address: data.businesses[0].location.display_address
+      }
```

Comment:
Is this later going to support showing all businesses?

Reply:
i don't think so but this was for the purpose of making sure react was actually communicating with the server